dev-utils/lint-tav-json.js

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+// Check that the ".ci/tav.json" file includes all the modules defined in
+// the various ".tav.yml" files. If not, then CI "TAV" runs are not testing
+// all modules.
+//
+// Also check that the product of "versions" and "modules" in tav.json does
+// not exceed the 256 GH Actions limit.
+// https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits
+
+const fs = require('fs')
+const path = require('path')
+
+const glob = require('glob')
+const yaml = require('js-yaml')
+
+const TOP = path.resolve(path.join(__dirname, '..'))
+const TAV_JSON_PATH = path.join('.ci', 'tav.json')
+
+// ---- mainline
+
+function main (argv) {
+  let numErrors = 0
+
+  // Find all module names in ".tav.yml" files.
+  const moduleNamesFromYaml = new Set()
+  const tavYmlPaths = glob.sync(
+    '**/.tav.yml',
+    { ignore: ['**/node_modules/**'] }
+  )
+  // console.log('tavYmlPaths:', tavYmlPaths)
+  tavYmlPaths.forEach(p => {
+    const tavCfg = yaml.load(fs.readFileSync(p, 'utf8'))
+    Object.keys(tavCfg).forEach(k => {
+      const v = tavCfg[k]
+      moduleNamesFromYaml.add(v.name || k)
+    })
+  })
+  // console.log('moduleNamesFromYaml: ', moduleNamesFromYaml)
+
+  // Find module names in ".ci/tav.json".
+  const moduleNamesFromJson = new Set()
+  const tavJson = JSON.parse(fs.readFileSync(path.join(TOP, TAV_JSON_PATH)))
+  tavJson.modules.forEach(m => {
+    m.split(',').forEach(moduleName => {
+      moduleNamesFromJson.add(moduleName)
+    })
+  })
+  // console.log('moduleNamesFromJson: ', moduleNamesFromJson)
+
+  // Matrix 256 limit.
+  const matrixSize = tavJson.versions.length * tavJson.modules.length
+  if (matrixSize > 256) {
```

Comment:
That's pretty neat! If it helps the fix could be refactoring the existing TAV to run in chunks as done with https://github.com/elastic/apm-agent-python/blob/1e38ec53115edc70c36c6485259733a8cde02ed9/.github/workflows/test.yml#L59-L81